src/read_pool.rs

```diff
@@ -108,17 +117,60 @@ impl<E: Engine> Runner for ReadPoolRunner<E> {
 
     fn end(&mut self, local: &mut Local<Self::TaskCell>) {
         self.inner.end(local);
+        self.flush_metrics();
         unsafe { destroy_tls_engine::<E>() }
     }
 }
 
-impl<E: Engine> ReadPoolRunner<E> {
-    pub fn new(engine: E, inner: FutureRunner) -> Self {
+impl<E: Engine, R: FlowStatsReporter> ReadPoolRunner<E, R> {
+    pub fn new(engine: E, inner: FutureRunner, reporter: R) -> Self {
         ReadPoolRunner {
             engine: Some(engine),
+            reporter,
             inner,
         }
     }
+
+    // Only flush metrics by tick
+    fn maybe_flush_metrics(&self) {
```

Comment:
Would `try_flush_metrics` be better?

Reply:
`try_` usually returns a `Result` while here we just ignore too frequent flushes

Reply:
fixed. PTAL again.